geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/collections/OrderStatisticTreeTest.java

```diff
@@ -0,0 +1,580 @@
+      try {
+        iterator2.next();
+      } catch (ConcurrentModificationException ex) {
+        fail("iterator2 should NOT have thrown an exception.");
+      }
+    } catch (ConcurrentModificationException ex) {
+      assertThatThrownBy(() -> iterator2.next())
+          .isInstanceOf(ConcurrentModificationException.class);
+    }
+  }
+
+
+  @Test
+  public void testIteratorConcurrentRemove() {
+    for (int i = 10; i < 20; ++i) {
+      set.add(i);
+      tree.add(i);
+    }
+
+    Iterator<Integer> iterator1 = set.iterator();
+    Iterator<Integer> iterator2 = tree.iterator();
+
+    for (int i = 0; i < 4; ++i) {
+      iterator1.next();
+      iterator2.next();
+    }
+
+    // None of them contains 2, should not change the modification count.
+    set.remove(2);
+    tree.remove(2);
+
+    iterator1.remove();
+    iterator2.remove();
+
+    iterator1.next();
+    iterator2.next();
+
+    set.remove(12);
+    tree.remove(12);
+
+    // Both of them should throw.
+    assertThatThrownBy(() -> iterator1.remove())
+        .isInstanceOf(ConcurrentModificationException.class);
+    assertThatThrownBy(() -> iterator2.remove())
+        .isInstanceOf(ConcurrentModificationException.class);
+  }
+
+  @Test
+  public void testConcurrentOrIllegalStateOnRemove() {
+    for (int i = 0; i < 10; ++i) {
+      set.add(i);
+      tree.add(i);
+    }
+
+    Iterator<Integer> setIterator = set.iterator();
+    Iterator<Integer> treeIterator = tree.iterator();
+
+    set.add(100);
+    tree.add(100);
+
+    assertThatThrownBy(() -> setIterator.remove()).isInstanceOf(IllegalStateException.class);
+    assertThatThrownBy(() -> treeIterator.remove()).isInstanceOf(IllegalStateException.class);
+  }
+
+  @Test
+  public void testConcurrentIterators() {
+    for (int i = 0; i < 10; ++i) {
+      set.add(i);
+      tree.add(i);
+    }
+
+    Iterator<Integer> iterator1a = set.iterator();
+    Iterator<Integer> iterator1b = set.iterator();
+    Iterator<Integer> iterator2a = tree.iterator();
+    Iterator<Integer> iterator2b = tree.iterator();
+
+    for (int i = 0; i < 3; ++i) {
+      iterator1a.next();
+      iterator2a.next();
+    }
+
+    iterator1a.remove();
+    iterator2a.remove();
+
+    assertEquals(iterator1b.hasNext(), iterator2b.hasNext());
+
+    assertThatThrownBy(() -> iterator1b.next()).isInstanceOf(ConcurrentModificationException.class);
+    assertThatThrownBy(() -> iterator2b.next()).isInstanceOf(ConcurrentModificationException.class);
+  }
+
+  @Test
+  public void testToArray() {
+    Random r = new Random();
+
+    for (int i = 0; i < 50; ++i) {
+      int num = r.nextInt();
+      set.add(num);
+      tree.add(num);
+    }
+
+    assertTrue(Arrays.equals(set.toArray(), tree.toArray()));
+  }
+
+  @Test
+  public void testToArrayGeneric() {
+    for (int i = 0; i < 100; ++i) {
+      set.add(i);
+      tree.add(i);
+    }
+
+    Integer[] array1before = new Integer[99];
+    Integer[] array2before = new Integer[99];
```

Comment:
Can these numbers be replaced with `set.size() - 1` to make it more clear what's happening here?

Reply:
Elucidated.